pre_commit_dbt/utils.py

```diff
@@ -415,3 +419,109 @@ def __call__(
                 result[key] = value
 
         setattr(namespace, self.dest, result)
+
+
+def find_file_with_suffix(filename, suffix, root_path):
+    for file in glob.glob(f"{str(root_path)}/**/{filename + '.' + suffix}"):
+        return Path(file)
+
+
+def clean_yml_path(yml_path):
+    double_slash_index = yml_path.find("//")
+    return yml_path[double_slash_index + 2 :]
+
+
+def get_related_sqls(
+    yml_path,
+    paths: Sequence[str],
+    nodes: Dict[str, Any],
+    include_ephemeral: bool = False,
+):
+    yml_path = Path(yml_path)
+    yml_path_parts = yml_path.parts
+
+    root_path = yml_path_parts[0]
+    dbt_patch_path = "/".join(yml_path_parts)
+
+    for key, node in nodes.items():
+        if (
+            not include_ephemeral
+            and node.get("config", {}).get("materialized") == "ephemeral"
+        ):
+            continue
+
+        if node["patch_path"] and dbt_patch_path in node["patch_path"]:
+            if ".sql" in node["original_file_path"]:
+                target_sql_name = f"{root_path}/{node['original_file_path']}"
+                if target_sql_name not in paths:
+                    paths.append(target_sql_name)
```

Comment:
You're modifying the parameter, so no need to return it again on line 459. But, are you sure you want to modify it instead of returning a new list?

Reply:
Newest commit will use `paths` only for looping, and a `paths_with_missing` that contains the original `*paths` and those appended on the "search" methods

I don't find appending to `paths_with_missing` being a contradictory case. Believe working it this way is the same as receiving results (via return) and re-interpreting list content after the function call (to see if they already exist or not)

Thought about working with Sets, but the original class expects receives a `Sequence[str]`, and for what I understood of Python docs, a Set is not a Sequence